### PR TITLE
chore: add dangerjs

### PR DESCRIPTION
**Pull request by alinurr1**
## Description of the change
Adding dangerjs for PR checks
Initial rule checks presence of changelog file

## Testing the change

- [ ] Write your testing instructions here

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.

### CODE REVIEW

**Comment by alpacatron** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>You don't have a changeset file for your modifications.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b55eb3e33fee8f32f676519d830a4caed3f836ac
</p>

**Comment by sliminas** [comment]
Isn't it dangerous to add a package with such a name? :smile:

File: .gitignore
Comment (apvale) [reply .gitignore:2]: It's from JetBrains IDEs. So, we need to add them into . gitignore.

**Review by apvale — APPROVED** [review]
I don't have issues in having a file called `dangerfile` since it seems the usual thing (?) while setting up this package.

File: .gitignore
Comment (alinurr1) [reply .gitignore:2]: it's a similar package to VSCode development cache files, that are generated by Webstorm

**Review by padulafacundo — APPROVED** [review]
I can see the comment from alpacatron, but I don't see the GH action or the danger step. Why is that? 🤔

**Comment by apvale** [comment]
This one, @padulafacundo ? 
image

**Comment by apvale** [comment]
@alinurr1 Can you update the yarn.lock file before merging this PR? It seems that the linter and tests didn't run properly in your PR. 👀

**Comment by alpacatron** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>You don't have a changeset file for your modifications.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a944b628c11b189167e4a368bff28e10a54f6bd6
</p>

**Comment by alpacatron** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>You don't have a changeset file for your modifications.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3a23f1cf6c3f6daf227906d151bc4c63ceeeaecc
</p>

File: dangerfile.ts
Comment (richbachman) [inline dangerfile.ts:20]: Do we need this warning message here?

Comment (alinurr1) [reply dangerfile.ts:20]: Added this line, just in case, if the files that were changed do not get into the first condition

File: yarn.lock
Comment (nyan07) [inline yarn.lock:2333]: I think it should change only the package you add, right?

File: dangerfile.ts
Comment (apvale) [reply dangerfile.ts:20]: Yeah, I agree with Rich in this case, Alinur. Maybe we don't need to warn when the PR doesn't have changesets. Besides that, now I'm wondering why we have this alert from danger if we already have this bot https://github.com/Localitos/pluto/pull/39#issuecomment-1257852368 🤔

Comment (richbachman) [reply dangerfile.ts:20]: The bot doesn't block a PR from being merged without a changeset, since some PRs won't need a changeset. Like overall repo infrastructure work.

Danger is in place to catch a human of merging PRs that need a changeset.

Right now, that warning will happen on every PR that doesn't include a changeset, which is a bit overkill. We only want an error if a package is new or edited, and is missing the changeset file.

Let me know if you have any questions.

Comment (apvale) [reply dangerfile.ts:20]: So, we can remove this bot, right? 

image

Comment (richbachman) [reply dangerfile.ts:20]: We need to keep the bot. We can just remove the warning on the lines in the original comment.

**Comment by richbachman** [comment]
@alinurr1 along with removing the warning in the messages above, will you also create a commit to this PR where you change something in a package without a changeset, so that we can verify this check fails?

Screen Shot 2022-09-27 at 11 23 04

After verifying you can revert the change to the package. Thanks!

**Comment by alinurr1** [comment]
@richbachman, I changed the file without adding the changeset and danger now blocks the PR, though I checked it by manually triggering danger from the terminal, this is because the PR is not merged yet to main to trigger danger CI automatically
Screen Shot 2022-09-28 at 13 36 58

**Comment by apvale** [comment]
Thanks, Alinur! It worked:

image

🙏🏻

**Pull request merged**
